.github/workflows/ubuntu-20.04.yml

```diff
@@ -0,0 +1,212 @@
+          # Set up VAL.
+          sudo apt-get -y install flex bison
+          git clone https://github.com/KCL-Planning/VAL.git
+          cd VAL
+          git checkout a5565396007eee73ac36527fbf904142b3077c74
+          make clean  # Remove old build artifacts and binaries.
+          sed -i 's/-Werror //g' Makefile  # Ignore warnings.
+          make -j8
+          mv validate ../
+          cd ../
+          ls -la
+          rm -rf VAL
+
+      - name: archive-files
+        if: ${{ matrix.compiler-version }} == gcc
+        run: |
+          cd ../
+          tar cfz ${{ matrix.compiler-version }}.tar.gz downward lib
+
+      - name: upload-files
+        if: ${{ matrix.compiler-version }} == gcc
+        uses: actions/upload-artifact@v1
+        with:
+          name: compiled-planner
+          path: /home/runner/work/downward/${{ matrix.compiler-version }}.tar.gz
+
+
+  test: # identifier of the job. Jobs run in parallel unless specified otherwise.
+    name: test
+    runs-on: ubuntu-20.04
+    needs: compile # TODO: actually, this only depends on compile with gcc
+    strategy:
+      matrix:
+        compiler-version: [gcc]
+        python-version: [3.6, 3.8]
+    steps: # each - is a new sequentially run step
+      - name: setup-python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: x64
+
+      - name: download-files
+        uses: actions/download-artifact@v1
+        with:
+          name: compiled-planner
+
+      - name: extract-files
+        # We need to make sure that paths are the same as in the first job,
+        # otherwise cmake exits with an error when called during tests.
+        # Alternatively, we could change the tests so that they don't build.
+        run: |
+          cd /home/runner/work/downward/downward/compiled-planner
+          tar xfz ${{ matrix.compiler-version }}.tar.gz
+          shopt -s dotglob
+          mv downward/* ../
+          mv lib/ ../../
+          cd ../
+          rm -r compiled-planner
+
+      - name: setup-dependencies
+        run: |
+          pip3 install pytest tox
+          sudo apt-get update
+          sudo apt-get -y install clang-tidy-8 valgrind zlib1g-dev libgmp3-dev
+          sudo apt-get -y install ${{ matrix.compiler-version }}
+
+      - name: test
+        env:
+          DOWNWARD_CPLEX_ROOT: /home/runner/work/downward/lib/ibm/ILOG/CPLEX_Studio129/cplex
+          DOWNWARD_SOPLEX_ROOT: /home/runner/work/downward/lib/soplex-3.1.1
+          DOWNWARD_COIN_ROOT: /home/runner/work/downward/lib/coin
+        run: |
+          export PATH="$(pwd):$PATH"  # Add uncrustify and VAL to PATH.
+          cd misc/
+          python_version=${{ matrix.python-version }}
+          if [[ $python_version=="3.6" ]]; then
+            tox -e py36,translator,search,style
+          elif [[ $python_version=="3.8" ]]; then
+            tox -e py38,translator,search,style
+          else
```

Comment:
We could remove the style test from one of the versions. The results should be the same.